### ISSUE TITLE
Crash on reading file

### ISSUE THREAD

**Issue by bubnenkoff** [jasonwhite/io#7]
It's seems that it's crush on reading file, because it's not-unicode.

See attach.
[apptest.zip](https://github.com/jasonwhite/io/files/408396/apptest.zip)

**Comment by jasonwhite** [comment]
Yep, this is definitely a bug. Thanks for the simple repro case.

The UTF decoder seems to be throwing an exception because its not receiving a full UTF sequence from the line splitter. I took a look at `std.stdio` to see how it handles this. It looks like it detects if a byte is part of a UTF sequence and keeps on reading until it has the full sequence. This might be the way to go. I'll have to look into this more.

**Comment by bubnenkoff** [comment]
So it would crush on every UTF file?
